projects/Mallard/src/components/article/html/render.ts

```diff
@@ -18,19 +18,42 @@ const styles = css`
         font-family: 'GuardianTextEgyptian-Reg';
         padding: ${metrics.vertical}px ${metrics.article.sides}px;
     }
-    #app p {
+    #app p, figure {
         margin-bottom: ${metrics.vertical * 2}px;
     }
     #app a {
         color: ${color.primary};
         text-decoration-color: ${color.line};
     }
+    figcaption {
+        padding-top: 5px;
+        font-size: ${12 * PixelRatio.getFontScale()}px;
+        color: #767676;
+        line-height: 1rem;
+    }
 `
 
+const renderMediaAtom = (mediaAtomElement: MediaAtomElement) => {
+    return `
+    <figure style="overflow: hidden;">
+        <iframe scrolling="no" src="https://embed.theguardian.com/embed/atom/media/${mediaAtomElement.atomId}" style="width: 100%; display: block;" frameborder="0"></iframe>
+        <figcaption>${mediaAtomElement.title}</figcaption>
+    </figure>`
+}
+
 export const render = (article: BlockElement[]) => {
     const html = article
-        .filter(el => el.id === 'html')
-        .map(el => (el as HTMLElement).html)
+        .filter(el => el.id === 'html' || el.id === 'media-atom')
+        .map(el => {
+            switch (el.id) {
+                case 'html':
+                    return (el as HTMLElement).html
+                case 'media-atom':
+                    return renderMediaAtom(el as MediaAtomElement)
```

Comment:
do we need this type casting? shouldn't the switch statement already refine the type down?

Reply:
Nice - didn't know TS was that clever - will change.

Reply:
I'm not sure if it is!! Give it a try

Reply:
it is.